lib/galaxy/managers/histories.py

```diff
@@ -207,6 +208,12 @@ def p_tag_filter(term_text: str, quoted: bool):
                 self.model_class.deleted == (true() if show_deleted else false())
             )
 
+        # By default, only return non-archived histories when we are showing the current user's histories
+        # if listing other users' histories, we don't filter out archived histories as they may be
+        # public or shared with the current user
+        if not show_archived and show_own:
```

Comment:
```suggestion
        if show_own and not show_archived:
```
This swap is such a minor change, but I feel it helps me understand the logic: i.e., then I read "if the user is viewing their own histories AND they did not select the option to also view their archived histories, do this." Does this make sense?

Reply:
Also, is it possible to have `not show_archived` AND `not show_own`? i.e., "I don't wan't to include other people's histories that they have archived". If such a use case does not exist (I can't imagine why one would need that), then maybe just checking the `not show_archived` is enough?

Reply:
All the logic here is "a bit" confusing. Still, the thing is, relaying only in `not show_archived`, will not include our own published (and archived) histories when displaying "All the public histories in the server", and adding the `show_archived=true` in this case, will mix all public with our own private archived histories... so, confusing :sweat: 

Which means this change would break this test:

https://github.com/galaxyproject/galaxy/pull/17856/files#diff-0e4282e792feb48bd4102022b4520bc3213f1afd5df0b78bbd440c20b01f617fR349

Reply:
Thanks for the explanation! It is indeed rather confusing, but I guess it's due to the inherent complexity of the logic behind selecting histories on these overlapping criteria.  Here's what I used to better understand what was going on; posting this here for future reference, in case this might be helpful:
![image](https://github.com/galaxyproject/galaxy/assets/6132258/ba786117-759a-48e1-bed6-5de7e87937f9)